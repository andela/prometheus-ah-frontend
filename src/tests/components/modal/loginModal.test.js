import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedLoginModal, { LoginModal } from '../../../components/modals/LoginModal';


const createMockStore = configureMockStore([thunk]);
const store = createMockStore({});

const props = {
  modal: { current: 'signUp' },
  onClick: jest.fn(),
  hideModal: jest.fn(),
  closeModal: jest.fn(),
};
describe('Sign Up Modal Component', () => {
  it('should ', () => {
    const wrapper = shallow(<LoginModal {...props} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.exists()).toBe(true);
  });

  it('should render connected component', () => {
    const wrapper = shallow(<ConnectedLoginModal store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
