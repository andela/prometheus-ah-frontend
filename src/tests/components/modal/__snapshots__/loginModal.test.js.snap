// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Modal Component should render connected component 1`] = `
<LoginModal
  closeModal={[MockFunction]}
  hideModal={[Function]}
  modal={
    Object {
      "current": "signUp",
    }
  }
  onClick={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
